### ISSUE TITLE
Segmentation Fault

### ISSUE THREAD

**Issue by kyleabeauchamp** [daler/pybedtools#168]
AFAIK pybedtools should be able to parse a bgzipped vcf, right?  On my system (OSX + anaconda2 + pybedtools 0.7.7 + bioconda + bgzip 1.3) the following leads to a segfault on loading the compressed vcf.  

```
wget https://raw.githubusercontent.com/daler/pybedtools/master/pybedtools/test/data/1000genomes-example.vcf
bgzip -c 1000genomes-example.vcf > 1000genomes-example.vcf.gz

ipython
import pybedtools
x = pybedtools.BedTool("1000genomes-example.vcf")
y = pybedtools.BedTool("1000genomes-example.vcf.gz")
# Segfault
```

I'm also trying to track down a related issue where changing the VCF version header from 4.0 to 4.1 leads to a segfault only on compressed (bgzip or gzip) VCFs.  

I've also tried tabixing the files, to no avail.

**Comment by daler** [comment]
Looks like it's `pysam`. What's happening is that `y = pybedtools.BedTool("1000genomes-example.vcf.gz")` triggers a call to `pybedtools.helpers.isBAM`, which in turn checks to see if it's BGZIPed. It correctly identifies that it's BGZIPed, and then tries to differentiate between an arbitrary BGZIP and an actual BAM file. The problem is that I had been doing this by catching the ValueError raised when trying to open the file with `pysam.Samfile`, but this is what causes the segfault.

E.g., 

```python
pysam.Samfile("1000genomes-example.vcf.gz")
# segfaults
```

The solution is to use the BAM magic number to detect if it's BAM or not, but I must not be reading the SAM spec correctly since I'm unable to extract it properly using `struct.unpack_from`. Will fix this shortly.

**Comment by daler** [comment]
This should be fixed now in e5105a8 -- at least [this test](https://github.com/daler/pybedtools/commit/e5105a87b822297202ae3a2edbf8410dc060272b#diff-e9ac3fd4ff0a367972edda9d92e0c999R2047) no longer segfaults.

**Comment by kyleabeauchamp** [comment]
Thanks!
